docs/sources/configure-server/about-server-api.md

```diff
@@ -61,55 +64,119 @@ foo;bar
     "units": "bytes",
     "aggregation": "average",
     "display-name": "inuse_space_bytes",
-    "sampled": false,
+    "sampled": false
+  },
+  "alloc_objects": {
+    "units": "objects",
+    "aggregation": "sum",
+    "display-name": "alloc_objects_count",
+    "sampled": true
+  },
+  "cpu": {
+    "units": "samples",
+    "aggregation": "sum",
+    "display-name": "cpu_samples",
+    "sampled": true
   },
   // pprof supports multiple profiles types in one file,
   //   so there can be multiple of these objects
 }
 ```
 
+Explanation of sample type configuration fields:
+
+- **units**
+  - Supported values: `samples`, `objects`, `bytes`
+  - Description: Changes the units displayed in the frontend. `samples` = CPU samples, `objects` = objects in RAM, `bytes` = bytes in RAM.
+- **display-name**
+  - Supported values: Any string.
+  - Description: This becomes a suffix of the app name, e.g., `my-app.inuse_space_bytes`.
+- **aggregation**
+  - Supported values: `sum`, `average`.
+  - Description: Alters how data is aggregated on the frontend. Use `sum` for data to be summed over time (e.g., CPU samples, memory allocations), and `average` for data to be averaged over time (e.g., memory in-use objects).
+- **sampled**
+  - Supported values: `true`, `false`.
+  - Description: Determines if the sample rate (specified in the pprof file) is considered. Set to `true` for sampled events (e.g., CPU samples), and `false` for memory profiles.
+
+This configuration allows for customized visualization and analysis of various profile types within Pyroscope.
+
+### Uploading a pprof File to Pyroscope Server Using profilecli
+
+Using `profilecli`, you can easily upload your pprof files to the Pyroscope server. `profilecli` is a command-line utility that simplifies the process by allowing you to directly upload profiles with a single command. Below is a guide on how you can use `profilecli` for uploading a pprof file:
+
+#### Prerequisites
+- Ensure you have `profilecli` installed on your system.
+- Have the pprof file you want to upload ready.
+
+#### Steps to Upload
+
+1. **Identify the pprof file and Pyroscope server details.**
+
+   - Path to your pprof file: `path/to/your/pprof-file.pprof`
+   - Pyroscope server URL: If using Grafana Cloud, for example, `https://profiles-prod-001.grafana.net`. For local instances, it could be `http://localhost:4040`.
+
+2. **Determine Authentication Details (if applicable).**
+
+   - If using cloud or authentication is enabled on your Pyroscope server, you will need the following:
+     - Username: `<username>`
+     - Password: `<password>`
+
+3. **Specify any Extra Labels (Optional).**
+
+   - You can add additional labels to your uploaded profile using the `--extra-labels` flag.
+
+4. **Construct and Execute the Upload Command.**
+
+   - Here's a basic command template:
+     ```
+     ./profilecli upload --url=<pyroscope_server_url> --username=<username> --password=<password> path/to/your/pprof-file.pprof
```

Comment:
Maybe we should be adding a service_name label so the app UI can handle it?

Potentially we should maybe also just do that as default in profilecli?

Reply:
since we don't have docs not actually sure where that goes / how to add it can you make a commit here to add it?

Reply:
Created an issue for this https://github.com/grafana/pyroscope/issues/2642

Reply:
merged this so @aleks-p can update docs in his PR where this is added